src/scripts/underground/Underground.ts

```diff
@@ -174,7 +174,7 @@ class Underground implements Feature {
         const item = UndergroundItems.getById(id);
 
         if (item.valueType == UndergroundItemValueType.EvolutionItem) {
-            const evostone: EvolutionStone = (ItemList[item.valueType] as EvolutionStone);
+            const evostone: EvolutionStone = (ItemList[item.itemListName] as EvolutionStone);
```

Comment:
Instead of adding a new variable, couldn't we just do something like this instead?
```suggestion
            const evostone: EvolutionStone = (ItemList[StoneType[item.type]] as EvolutionStone);
```

Reply:
It looks like it works! Have cleaned up my solution, so ready for review again